bookrack-backend/src/main/java/org/example/bookrackbackend/BookController.java

```diff
@@ -29,18 +29,21 @@ public Book getBookById(@PathVariable String id){
     }
 
     @PostMapping()
-    public Book addBook(@RequestPart BookCreationDTO bookDTO, @RequestPart MultipartFile image) throws IOException {
-        return bookService.addBook(bookDTO,image);
+    public Book addBook(@RequestParam("file") MultipartFile image,@RequestParam("title") String title,
+                        @RequestParam("author") String author,
+                          @RequestParam("country") String country, @RequestParam("year")
```

Comment:
is @RequestParam batter than @RequestBody??

Reply:
RequestParam is used for extracting query parameters/form data from the request URL (which is what I am doing here). RequestBody on the other hand is used to extract (JSON) data from the body of a HTTP request and convert this to a Java object.